datafusion/functions/Cargo.toml

```diff
@@ -47,9 +49,22 @@ path = "src/lib.rs"
 
 [dependencies]
 arrow = { workspace = true }
+arrow-array = { workspace = true }
 base64 = { version = "0.21", optional = true }
+chrono = { workspace = true }
 datafusion-common = { workspace = true }
 datafusion-execution = { workspace = true }
 datafusion-expr = { workspace = true }
 hex = { version = "0.4", optional = true }
+itertools = { workspace = true }
 log = "0.4.20"
+rand = { workspace = true }
```

Comment:
I think several of these dependencies are only used in the tests (so they could be dev depenencies). I'll make a follow on PR clean it up as this one is already really large.